src/diffusers/loaders.py

```diff
@@ -1645,7 +1667,20 @@ def lora_scale(self) -> float:
         return self._lora_scale if hasattr(self, "_lora_scale") else 1.0
 
     def _remove_text_encoder_monkey_patch(self):
-        self._remove_text_encoder_monkey_patch_classmethod(self.text_encoder)
+        if self.use_peft_backend:
+            remove_method = recurse_remove_peft_layers
+        else:
+            warnings.warn(
```

Comment:
```suggestion
            deprecate
```

Can you please use the deprecate function here? For an example, see: https://github.com/huggingface/diffusers/blob/e312b2302b5445271198ebed8f2fbcd543633f78/src/diffusers/loaders.py#L2121